[openstack_instack] Collect undercloud debug information

stuggi opened the pull request:
If auth information got sourced before sosreport run
(OS_USERNAME, OS_PASSWORD, OS_TENANT_NAME) collect
information from nova, neutron, keystone, glance, ironic
and heat.
Only undercloud specific heat information is collected in
instack plugin.

Signed-off-by: Martin Schuppert <mschuppe@redhat.com>

bmr-cymru commented on sos/plugins/openstack_glance.py:
```diff
@@ -44,6 +46,20 @@ def setup(self):
+        if run_cmds:
+            self.add_cmd_output(
+                "openstack image list --long",
+                suggest_filename="openstack_image_list--long")
```
Don't set `suggest_filename` unless you are setting it to something other than the default mangled command name.

bmr-cymru commented on sos/plugins/openstack_heat.py:
```diff
@@ -32,6 +34,20 @@ def setup(self):
+        if run_cmds:
+            self.add_cmd_output(
+                "openstack stack list",
+                suggest_filename="openstack_stack_list")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_instack.py:
```diff
@@ -28,6 +30,41 @@ def setup(self):
+            # get status of overcloud stack and resources
+            self.add_cmd_output(
+                "openstack stack show overcloud",
+                suggest_filename="openstack_stack_show_overcloud")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_instack.py:
```diff
@@ -28,6 +30,41 @@ def setup(self):
+                suggest_filename="openstack_stack_show_overcloud")
+            self.add_cmd_output(
+                "openstack stack resource list -n 10 overcloud",
+                suggest_filename="openstack_stack_resource_list-n10_overcloud")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_instack.py:
```diff
@@ -28,6 +30,41 @@ def setup(self):
+                        cmd = "openstack software deployment show " \
+                                "--long %s" % (deployment)
+                        self.add_cmd_output(
+                            cmd,
```
Why does this need to be on a newline? Only lists inside `[..]` use the bracket-then-newline style in sos unless there's no space on the line.

If that makes the line with the `kwarg` harder to format I'd suggest using a variable for the file name - building them in-line is always a bit tricky to format readably since the argument name is already so long.

bmr-cymru commented on sos/plugins/openstack_ironic.py:
```diff
@@ -37,6 +39,23 @@ def setup(self):
+        if run_cmds:
+            self.add_cmd_output(
+                "openstack baremetal node list --long",
+                suggest_filename="openstack_baremetal_node_list--long")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_ironic.py:
```diff
@@ -37,6 +39,23 @@ def setup(self):
+            self.add_cmd_output(
+                "openstack baremetal port list",
+                suggest_filename="openstack_baremetal_port_list")
+
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_keystone.py:
```diff
@@ -41,6 +43,23 @@ def setup(self):
+        if run_cmds:
+            self.add_cmd_output(
+                "openstack endpoint list",
+                suggest_filename="openstack_endpoint_list")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_keystone.py:
```diff
@@ -41,6 +43,23 @@ def setup(self):
+                suggest_filename="openstack_endpoint_list")
+            self.add_cmd_output(
+                "openstack catalog list",
+                suggest_filename="openstack_catalog_list")
```
Same thing for `suggest_filename` here.

bmr-cymru commented on sos/plugins/openstack_neutron.py:
```diff
@@ -36,6 +38,26 @@ def setup(self):
+        if run_cmds:
+            self.add_cmd_output(
+                "openstack subnet list",
+                suggest_filename="openstack_subnet_list")
```
Same thing for `suggest_filename` here.

bmr-cymru replied:
(and the next two in this file)

bmr-cymru commented:
pep8 no likey:

```
$ pep8 sos
sos/plugins/openstack_instack.py:62:33: E127 continuation line over-indented for visual indent
```

stuggi commented:
Thanks, yes I should have not set explicitly the file names . I think I have addressed them all. Also added mistral and zaqar to the instack plugin, but missed to mention this in the commit message. Let me know if I should change the commit message.

pmoravec commented on sos/plugins/openstack_glance.py:
```diff
@@ -44,6 +46,18 @@ def setup(self):
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
+                run_cmds = True
```
Do you want to set run_cmds to True if and only if *all* the three variables are set? If so, then this wont work.. Check os.environ containing just one of the three vars - you get two warnings but run_cmds=True.

(or is your intent to set run_cmds if at least one such variable is set? that's what the code does..)

bmr-cymru replied:
You can also avoid the explicit loop entirely by using a list comprehension and `all()`:
```python
vars = [p in os.environ for p in ['OS_USERNAME', 'OS_PASSWORD', 'OS_TENANT_NAME']]
if not all(vars):
    # freak out
```

stuggi replied:
Oh yes, I blindly c/p it from the existed nova plugin. I'll fix this and submit an updated patch. Also I will check the other openstack plugins we already have. I know e.g. the same is in gnocchi.

pmoravec commented on sos/plugins/openstack_heat.py:
```diff
@@ -32,6 +34,18 @@ def setup(self):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
Ditto as above. If the intention is to set run_cmds to True only if all three vars are set, then try e.g.:

    run_cmds = True
    for os_var ..:
        if os_var not in os.environ:
            run_cmds = False
            self.soslog.warning(..)

pmoravec commented on sos/plugins/openstack_instack.py:
```diff
@@ -28,6 +30,50 @@ def setup(self):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
ditto as above

pmoravec commented on sos/plugins/openstack_ironic.py:
```diff
@@ -37,6 +39,19 @@ def setup(self):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
ditto

pmoravec commented on sos/plugins/openstack_keystone.py:
```diff
@@ -41,6 +43,19 @@ def setup(self):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
ditto

pmoravec commented on sos/plugins/openstack_neutron.py:
```diff
@@ -36,6 +38,20 @@ def setup(self):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
ditto

pmoravec commented on sos/plugins/openstack_nova.py:
```diff
@@ -28,17 +29,18 @@ class OpenStackNova(Plugin):
+                self.soslog.warning("%s not found in environment variables"
+                                    " which is required to run commands"
+                                    % (os_var))
+            else:
```
ditto

stuggi commented:
changes implemented. also submitted a fix for gnocchi in https://github.com/sosreport/sos/pull/956

bmr-cymru commented on sos/plugins/openstack_glance.py:
```diff
@@ -2,6 +2,7 @@
 # Copyright (C) 2012 Rackspace US, Inc.,
```
The commit subject is misleading - if you are going to change all the OpenStack plugins in one patch, use `[openstack_*]` as the commit tag. Otherwise you should never e.g. have hunks that apply to `openstack_glance.py` in a patch with a subject of `[openstack_instack]` - it's just confusing for anyone trying to understand the logs down the line.

Your other option here would be to make one commit per file - this is especially preferred if you are making other sets of unrelated changes (e.g. the very large addition to `openstack_insight.py` vs. the relatively minore env. var. change in all the others).

I'm not going to make you respin and split up the patch series at this stage, but it is the preferred style in `sos` (and many other projects), but the commit subject definitely needs fixing before this can go in.

stuggi replied:
will split in furute

bmr-cymru commented on sos/plugins/openstack_nova.py:
```diff
@@ -28,44 +29,37 @@ class OpenStackNova(Plugin):
+                cmd = "openstack server show %s" % (instance)
+                self.add_cmd_output(
+                    cmd,
+                    suggest_filename="instance-" + instance + ".log")
```
Ditto these changes - would have been much better to do the code shuffling in a separate patch.

stuggi commented:
changed commit subject to openstack_* . Will split such change in future to different patches.

bmr-cymru commented:
Thanks! I'll get these into this week's review batch.

bmr-cymru commented:
Cheers - these look fine now. Will hit master in a moment.
